### ISSUE TITLE
Publisher: Hosts filtering does not affect AttrDefs

### ISSUE THREAD

**Issue by jakubjezek001** [ynput/OpenPype#4726]
**Running version**
3.15.3-nightly.4

**Describe the bug**
Hosts filtering on an integration plugin, which is having defined plugin attributes, are showing even hosts are not mentioning a particular host.

**To Reproduce**
following example is taken form https://github.com/ynput/OpenPype/pull/4715
1. Open settings at `project_settings/deadline/publish/SuspendedPublishingJobOnFarm/hosts` and remove all hosts so the field is empty.
2. Save settings 
3. Open Nuke in your testing workfile
4. hit Publish and click to any `render `or `prerender `family instance to see right panel plugin attributes.
5. Notice that **Suspended publish** toggle is still showing.

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/40640033/228209755-4e5b8d05-809e-40da-bd7f-6a61b4337cf3.png)

![image](https://user-images.githubusercontent.com/40640033/228209653-8142171d-41de-4f95-90f2-54f31885024f.png)

**Desktop (please complete the following information):**
 - OS: [e.g. windows]
 - Host: [e.g. Maya, Nuke, Houdini]

[cuID:[OP-5447](https://app.clickup.com/t/863gbexbz)]

**Comment by BigRoy** [comment]
I believe with the new publisher you actually explicitly need to set the settings using the `apply_settings` method on the class. It doesn't automatically find a key in settings that matches the class name like the old logic to 'automatically' apply the settings. Right @iLLiCiTiT ?

**Comment by iLLiCiTiT** [comment]
You're right but the issue is that attribute definitions from pyblish plugins are shown even if plugins' host filters do not match registered hosts.

**Comment by BigRoy** [comment]
> You're right but the issue is that attribute definitions from pyblish plugins are shown even if plugins' host filters do not match registered hosts.

Let's clarify that in the issue title?
